source/elements/oneDAL/source/algorithms/decomposition/pca.rst

```diff
@@ -1,3 +1,447 @@
+   spec does not require implementation of this sign-flip technique.
+   Implementers can choose an arbitrary technique that modifies sign of the
+   eigenvectors or use deterministic algorithm for eigenvectors computations.
+
+
+-------------
+Usage example
+-------------
+::
+
+   onedal::pca::model run_training(const onedal::table& data) {
+
+      const auto pca_desc = onedal::pca::desc<float>{}
+         .set_component_count(5)
+         .set_is_deterministic(true);
+
+      const auto result = onedal::train(pca_desc, data);
+
+      print_table("means", result.get_means());
+      print_table("variances", result.get_variances());
+      print_table("eigenvalues", result.get_eigenvalues());
+      print_table("eigenvectors", result.get_model().get_eigenvectors());
+
+      return result.get_model();
+   }
+
+
+::
+
+   onedal::table run_inference(const onedal::pca::model& model,
+                               const onedal::table& new_data) {
+
+      const auto pca_desc = onedal::pca::desc<float>{}
+         .set_component_count(model.get_component_count());
+
+      const auto result = onedal::infer(pca_desc, model, new_data);
+
+      print_table("labels", result.get_transformed_data());
+   }
+
+
+---
+API
+---
+Methods
+-------
+::
+
+   namespace method {
+      struct cov {};
+      struct svd {};
+      using by_default = svd;
+   } // namespace method
+
+
+.. namespace:: onedal::pca::method
+
+.. struct:: cov
+
+   Tag-type that denotes `Covariance-based method`_.
+
+
+.. struct:: svd
+
+   Tag-type that denotes `SVD-based method`_.
+
+
+.. type:: by_default = svd
+
+   Alias tag-type for the `SVD-based method`_.
+
+
+Descriptor
+----------
+::
+
+   template <typename Float = float,
+             typename Method = method::by_default>
+   class desc {
+   public:
+      desc();
+
+      int64_t get_component_count() const;
+      bool get_is_deterministic() const;
+
+      desc& set_component_count(int64_t);
+      desc& set_is_deterministic(bool);
+   };
+
+.. namespace:: onedal::pca
+
+.. class:: template<typename Float = float, \
+                    typename Method = method::by_default> \
+           desc
+
+   :tparam Float: The floating-point type that the algorithm uses for
+                  intermediate computations. Can be :expr:`float` or :expr:`double`.
+
+   :tparam Method: Tag-type that specifies variant of PCA algorithm. Can be
+                   :expr:`method::cov`, :expr:`method::svd` or :expr:`method::by_default`.
+
+   .. function:: desc()
+
+      Creates new instance of descriptor with the default attribute values.
+
+
+   .. member:: std::int64_t component_count = 0
+
+      The number of principal components :math:`r`. If it is zero, the algorithm
+      will compute the eigenvectors for all features, :math:`r = p`.
+
+      Getter & Setter
+         | ``std::int64_t get_component_count() const``
+         | ``desc& set_component_count(std::int64_t)``
+
+      Invariants
```

Comment:
what if r>p?

Reply:
It's not invariant for descriptor, but `r <= p` is precondition for training/inference.